coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -105,27 +107,66 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
                                    qContext: QueryContext,
                                    forceInProcess: Boolean = false): PlanResult = {
     // Should avoid this asInstanceOf, far many places where we do this now.
-    val params = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
     // MultiPartitionPlanner has capability to stitch across time partitions, however, the logic is mostly broken
     // and not well tested. The logic below would not work well for any kind of subquery since their actual
     // start and ends are different from the start/end parameter of the query context. If we are to implement
     // stitching across time, we need to to pass proper parameters to getPartitions() call
-    val paramToCheckPartitions = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
-    val partitions = getPartitions(logicalPlan, paramToCheckPartitions)
-
-    if (isSinglePartition(partitions)) {
-
-      val (partitionName, startMs, endMs, grpcEndpoint) = partitions.headOption match {
+    if (forceInProcess) {
+      // If inprocess is required, we will rely on the DefaultPlanner's implementation as the expectation is that the
+      // raw series is doing a remote call to get all the data.
+      logicalPlan match {
+        case lp: RawSeries    if lp.supportsRemoteDataCall=>
+            //println(lp)
+           // materialize the raw query for the partition by rewriting the prom parameters
+            // A Raw series from [x, y] offset o and lookback l is same as doing a raw export as of time instant y
+           // with a lookback of (y - x) + o + l, for each partition this internal spans, we will find the remote/local
+           // partitions and stitch the results
+            val params = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+            val rs = lp.rangeSelector.asInstanceOf[IntervalSelector]
+
+            val (rawExportStart, rawExportEnd) =
+              (rs.from - lp.offsetMs.getOrElse(0L) - lp.lookbackMs.getOrElse(0L), rs.to - lp.offsetMs.getOrElse(0L))
+
+            val partition = getPartitions(lp, params)
+            // For each partition, do a raw data export range query
+            val execPlans = partition.map(pa => {
+              val (thisPartitionStartMs, thisPartitionEndMs) =
+                (Math.max(pa.timeRange.startMs, rawExportStart), Math.min(pa.timeRange.endMs, rawExportEnd))
+              val totalOffsetThisPartitionMs = thisPartitionEndMs - thisPartitionStartMs
+              // Add additional 1 second to avoid ms to sec rounding issues
+              val thisPartitionLp = lp.copy(offsetMs = None, lookbackMs = Some(totalOffsetThisPartitionMs + 1000L))
```

Comment:
Should the time parameters be updated, too? Also, why is `offsetMs` forced to `None` here?

Reply:
I'll look at the time parameter. The offset is converted to the range of raw data that we are interested in and hence its made ``None``